src/sentry/rules/actions/notify_event_service.py

```diff
@@ -50,21 +75,41 @@ def after(self, event, state):
             self.logger.info('rules.fail.is_configured', extra=extra)
             return
 
-        plugin = plugins.get(service)
-        if not plugin.is_enabled(self.project):
-            extra['project_id'] = self.project.id
-            self.logger.info('rules.fail.is_enabled', extra=extra)
-            return
-
-        group = event.group
-
-        if not plugin.should_notify(group=group, event=event):
-            extra['group_id'] = group.id
-            self.logger.info('rule.fail.should_notify', extra=extra)
-            return
-
-        metrics.incr('notifications.sent', instance=plugin.slug)
-        yield self.future(plugin.rule_notify)
+        app = None
+        try:
+            app = SentryApp.objects.get(slug=service)
```

Comment:
see @mnoble's comment https://github.com/getsentry/sentry/pull/10662#issuecomment-442152098

Reply:
when plugins/apps get added/removed, do rules change? what happens if I install the pushover plugin, create a rule, uninstall it and install the pushover installation? don't i need to know that the rule was created for the plugin not the app?

Reply:
Rules don't change - so the slug will still remain. The idea would be if you have the old plugin enabled, when you install the new plugin, we would disabled the old plugin but since the slug is the same, the rule will still work just like before. And then we wouldn't let you enabled the old plugin unless you uninstall the new service.